bin/https.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <s2n.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+
+#include "error/s2n_errno.h"
+#include "stuffer/s2n_stuffer.h"
+#include "utils/s2n_safety.h"
+
+#define STRING_LEN 1024
+static char str_buffer[STRING_LEN];
+static s2n_blocked_status blocked;
+
+#define SEND(...) do { \
+    sprintf(str_buffer, __VA_ARGS__); \
+    GUARD(s2n_send(conn, str_buffer, strlen(str_buffer), &blocked)); \
+} while (0)
+
+#define BUFFER(...) do { \
+    sprintf(str_buffer, __VA_ARGS__); \
+    GUARD(s2n_stuffer_write_bytes(&stuffer, (const uint8_t *)str_buffer, strlen(str_buffer))); \
```

Comment:
nit: Might be worth adding something like `s2n_stuffer_printf` to stuffer lib in future.